presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/TransformExistsApplyToLateralNode.java

```diff
@@ -86,26 +105,68 @@ public Result apply(ApplyNode parent, Captures captures, Context context)
             return Result.empty();
         }
 
-        Symbol count = context.getSymbolAllocator().newSymbol(COUNT.toString(), BIGINT);
+        Optional<PlanNode> nonDefaultAggregation = rewriteToNonDefaultAggregation(parent, context);
+        return nonDefaultAggregation
+                .map(Result::ofPlanNode)
+                .orElseGet(() -> Result.ofPlanNode(rewriteToDefaultAggregation(parent, context)));
+    }
+
+    private Optional<PlanNode> rewriteToNonDefaultAggregation(ApplyNode parent, Context context)
+    {
+        checkState(parent.getSubquery().getOutputSymbols().isEmpty(), "Expected subquery output symbols to be pruned");
+
         Symbol exists = getOnlyElement(parent.getSubqueryAssignments().getSymbols());
+        Symbol subqueryTrue = context.getSymbolAllocator().newSymbol("subqueryTrue", BOOLEAN);
+
+        Assignments.Builder assignments = Assignments.builder();
+        assignments.putIdentities(parent.getInput().getOutputSymbols());
+        assignments.put(exists, new CoalesceExpression(ImmutableList.of(subqueryTrue.toSymbolReference(), BooleanLiteral.FALSE_LITERAL)));
+
+        PlanNode subquery = new ProjectNode(
+                context.getIdAllocator().getNextId(),
+                new LimitNode(
+                        context.getIdAllocator().getNextId(),
+                        parent.getSubquery(),
+                        1L,
+                        false),
+                Assignments.of(subqueryTrue, TRUE_LITERAL));
 
-        return Result.ofPlanNode(
+        PlanNodeDecorrelator decorrelator = new PlanNodeDecorrelator(context.getIdAllocator(), context.getLookup());
```

Comment:
why not to return lateral node here and leave decorrelation for another rule? When this rewrite does not work (when `rewriteToDefaultAggregation` is needed)?